host_source.go

```diff
@@ -260,6 +260,12 @@ func (h *HostInfo) DataCenter() string {
 	return dc
 }
 
+func (h *HostInfo) SetDatacenter(dc string) {
```

Comment:
> Generally, to avoid conflicts with upstream, would it be better to add this in a file called host_source_scylla.go?

moved to seprate file

> Is this something we can ship upstream?

Upstream doesn't use this function, so i guess not.